narwhals/_arrow/dataframe.py

```diff
@@ -602,31 +602,35 @@ def is_duplicated(self: Self) -> ArrowSeries:
 
         from narwhals._arrow.series import ArrowSeries
 
-        df = self._native_frame
+        index_token = generate_temporary_column_name(n_bytes=8, columns=self.columns)
+        df = self.with_row_index(index_token)._native_frame
 
         columns = self.columns
         col_token = generate_temporary_column_name(n_bytes=8, columns=columns)
         row_count = (
             df.append_column(col_token, pa.repeat(pa.scalar(1), len(self)))
-            .group_by(columns)
+            .group_by([x for x in columns if x != index_token])
```

Comment:
`columns` in line 608 should still be the original frame columns as it is from `self` and not from the newly created `df`